### PR TITLE
Add parallax effect to the header

### LINKED ISSUE #17: Add parallax effect to the header image

### PR DESCRIPTION
**Pull request by Polostor**
# Short description

Work in progress of having Parallax effect on the website as requested in #17.

Closes #17.

# Changes

feat: use magic scrolling
fix: fix missing github icon
feat(static): add more layers
feat: add background layers in idraw

# Static visuals

## Macbook
trim 692BC851-7156-4777-94AA-505ECA9C0CE4

## iPhone
trim 3AAE231D-54F9-4D94-B641-85030E565D10

### CODE REVIEW

**Comment by Polostor** [comment]
trim 692BC851-7156-4777-94AA-505ECA9C0CE4

trim 3AAE231D-54F9-4D94-B641-85030E565D10

**Comment by Polostor** [comment]
@Indy2222 it is **work in progress** but I'd like your thoughts on design.

File: sass/github.scss [hunk 1]
```diff
@@ -1,25 +1,35 @@
+#github {
```
Comment (Indy2222) [inline sass/github.scss:1]: There is a lot of unrelated whitspace / indentation changes. Can you remove this changes from the PR?

**Comment by Polostor** [comment]
@Indy2222 it should be done. What do you think? 🤔

File: templates/base.html [hunk 2]
```diff
@@ -33,14 +33,73 @@
...
+      layer.style.webkitTransform = "translate3d(0, " + offset + "px, 0)";
+      layer.style.MozTransform = "translate3d(0, " + offset + "px, 0)";
+      layer.style.msTransform = "translateY(" + offset + "px)";
+      layer.style.OTransform = "translate3d(0, " + offset + "px, 0)";
+      layer.style.transform = "translate3d(0, " + offset + "px, 0)";
+    }
+
+    ///
+    function OnScroll(e) {
+      if (!isios) {
```
Comment (Indy2222) [inline templates/base.html]: Why is this condition here?

File: templates/base.html [hunk 3]
```diff
@@ -33,14 +33,73 @@
...
+      <div class="bg" id="midground">&nbsp;</div>
+      <div class="bg" id="foreground">&nbsp;</div>
     </div>
     <div class="middle">
       {% block content %} {% endblock %}
     </div>
+  <script>
+
+    ///
+    function MoveParallax() {
```
Comment (Indy2222) [inline templates/base.html:56]: I would simplify this by by introducing an array of layers and assuming that the layers are equidistant (with `layerDist` in between them). The movement speed is inversely proportional to distance and thus:

```
var foregroundSpeedFactor = 0.5;
var foregroundDist = X;  // TODO use proper X
var layerDist = Y;  // TODO use proper Y
```

And then for each `layerIndex` the offset is `var offset = (scrollPosition * foregroundSpeedFactor * foregroundDist) / (foregroundDist + layerIndex * layerDist)`.

File: static/header_wide.idraw
Comment (Indy2222) [inline static/header_wide.idraw:1]: What is this file?

**Review by Indy2222 — CHANGES_REQUESTED** [review]

**Comment by Indy2222** [comment]
I tested it in Firefox and it is sometimes smooth and sometimes lagging for some reason.

File: static/header_wide.idraw
Comment (Polostor) [reply static/header_wide.idraw:1]: This is original file for an background image split to the layers..

File: templates/base.html (hunk 2 above)
Comment (Polostor) [reply templates/base.html]: It was an advice by ChatGPT. Some stuff was failing on iOS, but as I tried it on iPhone I have not seen anything wrong.

Probably futile. Gonna remove it.

File: sass/github.scss (hunk 1 above)
Comment (Polostor) [reply sass/github.scss:1]: I see. Fixed.

File: templates/base.html (hunk 3 above)
Comment (Polostor) [reply templates/base.html:56]: I prefer to keep it this way. Do not want to really play with it much longer TBH.
If you wish, feel free to make a new PR to make it leaner 👍

**Comment by Polostor** [comment]
> I tested it in Firefox and it is sometimes smooth and sometimes lagging for some reason.

So I have checked what is wrong and it may be a problem between forcing a hardware acceleration in `translate3d` and without it `translateY` would fail because of too many layers.

https://stackoverflow.com/a/22336799/4745695

**Pull request merged**